pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -277,15 +277,17 @@ func (m *ManagerImpl) genericDeviceUpdateCallback(resourceName string, devices [
 
 		if utilfeature.DefaultFeatureGate.Enabled(features.ResourceHealthStatus) {
 			// compare with old device's health and send update to the channel if needed
+			podUID := ""
```

Comment:
My suggestion, if I am wrong, please correct me. Can we abstract a method, updatePodUID? It seems that the logic of the two is similar. If a podUID is initialized, I think it may make the code more complicated.
like: 
```bash
if utilfeature.DefaultFeatureGate.Enabled(features.ResourceHealthStatus) {
    // updatePodUID is func to update podUID for the device
    updatePodUID := func(deviceID string) {
        podUID, _ := m.podDevices.getPodAndContainerForDevice(deviceID)
        if podUID != "" {
            podsToUpdate.Insert(podUID)
        }
    }

    if oldDev, ok := oldDevices[dev.ID]; ok {
        if oldDev.Health != dev.Health {
            updatePodUID(dev.ID)
        }
    } else {
        updatePodUID(dev.ID)
    }
}
```

Reply:
Looks like good advice, thanks~